src/Interpreters/Access/InterpreterCreateUserQuery.cpp

```diff
@@ -38,6 +39,9 @@ namespace
         else if (query.names->size() == 1)
             user.setName(query.names->front()->toString());
 
+        if (user.getName() == USER_INTERSERVER_MARKER || user.getName().empty())
+            throw Exception(ErrorCodes::BAD_ARGUMENTS, "User name is reserved");
```

Comment:
`USER_INTERSERVER_MARKER` is only a marker in TCP protocol, it's quite a random string which we expect no real user will use as their name. I'd prefer not to infect every user-handling code with this TCP protocol's internal details.

Reply:
Or at least let's do that in one place, for example we could make a new function `User::setName()` and check that the name is appropriate there.

Reply:
> It's highly unlikely that someone will make a user called " INTERSERVER SECRET "

But what if someone will try? It makes me ask questions like:
 - What will happen if someone create such user somehow?
 - Ok, it's forbidden to `CREATE USER` with spaces and not possible to specify it in xml, but can we get username like this from kerberos or ldap or other service [that will be added in future]?
 - What if someone will change parser to support username with spaces? (because there's no good reason for this restriction)
 - What if someone will add support for query parameters in `CREATE USER` query? (so it will be possible to pass arbitrary identifier)

I'd prefer not to think about it and not to answer these questions and simply add explicit checks for reserved usernames. (however, it's not clear if I need to add similar check to other places...)

Reply:
Ok, let's at least do to that in one place. You could define a new function User::setName() and check that the name is not empty and not the interserver secret. Please write a comment there why it's so. And this way it would be unnecessary to check that the name doesn't start with a space.